### PR TITLE
add Stargaze blockchain

### PR DESCRIPTION
**Pull request by jhernandezb**
This PR adds Stargaze a Cosmos SDK based blockchain. [Cosmos SDK](cosmos.network)  chains usually have the same APIs to retrieve the list of validators and voting power and I have abstracted the work made for cosmos hub chain to just point to the specific chain endpoint and name.

Screenshot 2023-08-16 at 9 24 15 AM

### About Stargaze
Stargaze is "The Interchain" NFT marketplace being the first interoperable layer-1 chain for NFTs. Stargaze is the app specific chain for NFTs in the Cosmos.

https://stargaze.zone
https://twitter.com/StargazeZone

### CODE REVIEW

File: core/chains/cosmos.go [hunk 1]
```diff
@@ -25,16 +26,20 @@ type cosmosResponse struct {
 }
 
 func Cosmos() (int, error) {
+	url := "https://proxy.atomscan.com/cosmoshub-lcd/cosmos/staking/v1beta1/validators?page.offset=1&pagination.limit=500&status=BOND_STATUS_BONDED"
+	return fetchCosmosSDKNakamotoCoefficient("cosmos", url)
+}
+
+func fetchCosmosSDKNakamotoCoefficient(chainName, url string) (int, error) {
```
Comment (xenowits) [inline core/chains/cosmos.go:33]: can rename to a shorter name `fetchCosmosNakamotoCoeff`

Comment (xenowits) [inline core/chains/cosmos.go:33]: can rename to a shorter name
```suggestion
// fetchCosmosSDKNakaCoeff returns the nakamoto coefficient for the provided cosmos SDK-based chain using the provided url.
func fetchCosmosSDKNakaCoeff(chainName, url string) (int, error) {
```

**Review by xenowits — APPROVED** [review]
Left a minor comment, but overall looks good. Thanks for your contribution 🚀

File: core/chains/cosmos.go (hunk 1 above)
Comment (jhernandezb) [reply core/chains/cosmos.go]: done

**Comment by jhernandezb** [comment]
Thanks for reviewing the PR I've addresses your comments

**Pull request merged**